addons/l10n_hk/models/res_bank.py

```diff
@@ -10,50 +10,64 @@
 class ResPartnerBank(models.Model):
     _inherit = 'res.partner.bank'
 
-    country_code = fields.Char(related='partner_id.country_code', string="Country Code")
-    l10n_hk_fps_type = fields.Selection([('id', "FPS ID"), ('mobile', "Mobile Number"), ('email', "Email Address")], string='FPS Type')
-    l10n_hk_fps_identifier = fields.Char(string="FPS Identifier", help="FPS ID, Mobile Number or Email Address")
+    proxy_type = fields.Selection(selection_add=[('id', "FPS ID"), ('mobile', "Mobile Number"), ('email', "Email Address")],
+                                  ondelete={'id': 'set default', 'mobile': 'set default', 'email': 'set default'})
 
-    @api.constrains('l10n_hk_fps_type', 'l10n_hk_fps_identifier')
-    def _check_l10n_hk_fps_identifier(self):
+    @api.constrains('proxy_type', 'proxy_value', 'partner_id')
```

Comment:
```suggestion
    @api.constrains('proxy_type', 'proxy_value', 'country_code')
```
Does it not work? It makes more sense to run the constrains when the country is changed since it is the field that matters here, and you could change the country without changing the partner

Reply:
Doesn't work when it switch the Account Holder from SG to HK